utils/workers/Creators.py

```diff
@@ -0,0 +1,364 @@
+        if self.combine or self.bitmask:
+            if len(labels) < 8:
+                self.dtype = np.uint8
+                self.bits - 8
+            elif len(labels) < 16:
+                self.dtype = np.uint16
+                self.bits = 16
+            elif len(labels) < 32:
+                self.dtype = np.uint32
+                self.bits = 32
+
+            elif len(labels) > self.max_labels:
+                log.warning(
+                    f"Due to the maximum integer length ({self.max_labels+1} bits), we can "
+                    f"only keep track of a maximum of {self.max_labels} ROIs with a bitmasked "
+                    f"combination. You have {len(labels)} ROIs."
+                )
+
+    def label2data(self, label, ohifviewer_info):
+        data = np.zeros(self.shape, dtype=np.bool)
+        for roi_type in ohifviewer_info:
+            for roi in ohifviewer_info[roi_type]:
+                if roi.get("location") == label:
+                    data = self.fill_roi_dicom_slice(
+                        data,
+                        roi["SOPInstanceUID"],
+                        roi["handles"],
+                        dicoms=self.dicoms,
+                        roi_type=roi_type,
+                    )
+
+        return data
+
+    def save_to_roi_dir(self, data):
+
+        data = data.astype(self.dtype)
+
+        for i, dicom_info in enumerate(self.dicoms.items()):
+            dicom_file = dicom_info[0]
+            dicom_data = dicom_info[1]
+            file_name = dicom_file.name
+            dicom_out = self.roi_dir / file_name
+
+            arr = data[:, :, i]
+            arr = arr.squeeze()
+
+            dicom_data.BitsAllocated = self.bits
+            dicom_data.BitsStored = self.bits
+            dicom_data.HighBit = self.bits - 1
+            # dicom_data['PixelData'].VR = "OW"
+            dicom_data.PixelData = arr.tobytes()
+            dicom_data.save_as(dicom_out)
+
+    @staticmethod
+    def fill_roi_dicom_slice(
+        data, sop, roi_handles, roi_type="FreehandRoi", dicoms=None, reactOHIF=True
+    ):
+
+        dicom_data = dicoms.values()
+        dicom_sops = [d.SOPInstanceUID for d in dicom_data]
+        slice = dicom_sops.index(sop)
+
+        swap_axes = True
+        flips = [False, False]
+
+        orientation_slice = data[:, :, slice]
+
+        if roi_type == "FreehandRoi":
+            if reactOHIF:
+                roi_points = roi_handles["points"]
+            else:
+                roi_points = roi_handles
+
+            # If this slice already has data (i.e. this label was used in an ROI
+            # perpendicular to the current slice) we need to have the logical or
+            # of that data and the new data
+
+            orientation_slice[:, :] = np.logical_or(
+                roi_tools.freehand2mask(
+                    roi_points, orientation_slice.shape, flips, swap_axes
+                ),
+                orientation_slice[:, :],
+            )
```

Comment:
Not sure it's important to explain, but I'm still not understanding this `logical_or` operation even with your comment :(

Reply:
If you have two ROI measurements of the same label that overlap, this creates a union mask of the two.